core/src/main/java/org/apache/iceberg/StaticTableScan.java

```diff
@@ -37,6 +38,28 @@ private StaticTableScan(TableOperations ops, Table table, Schema schema,
     this.buildTask = buildTask;
   }
 
+  /**
+   * Type of scan being performed by the buildTask, such as {@link MetadataTableType#HISTORY} when scanning
+   * a table's {@link org.apache.iceberg.HistoryTable}.
+   * <p>
+   * Used for logging and error messages.
+   */
+  protected String tableType() {
+    return "static";
```

Comment:
Rather than overriding this several places and using a generic string here to avoid making this class abstract, I suggest passing the `MetadataTableType` in through the constructor so this just returns `tableType.name()`.

Reply:
That makes sense. We're going to see a `table type static` in a notebook eventually this way and it's going to be not very comprehensible.

Unfortunately this method can't be abstract, so the constructor it is.